AudioKit/Common/Internals/AudioKit.swift

```diff
@@ -28,7 +28,31 @@ public typealias AKCallback = () -> Void
     // MARK: - Internal audio engine mechanics
 
     /// Reference to the AV Audio Engine
-    @objc open static var engine = AVAudioEngine()
+    @objc open static var engine = AVAudioEngine() {
+        didSet {
+            #if os(iOS)
+            if routeChangeObserver != nil { NotificationCenter.default.removeObserver(routeChangeObserver!) }
```

Comment:
Use if let rather than checking for nil and force unwrapping.
`if let rtc = routeChangeObserver { NotificationCenter.default.removeObserver(routeChangeObserver) }`